### PR TITLE
[bitnami/keycloak] Add dnsConfig to keycloak statefulset template

### LINKED ISSUE #16337: Add dnsConfig option to keycloak
### Name and Version

bitnami/keycloak 14.5.0

### What is the problem this feature will solve?

using dnsConfig, we can modify resolv.conf

### What is the feature you are proposing to solve the problem?

we can add ndots using dnsConfig, this will solve name resolution issues, if ndots >= 5 is used. Example, 
keycloak.mynamespace.svc.cluster.local can't be resolved if ndots is >=5. we can update ndots using dnsConfig in statefulset.

### What alternatives have you considered?

_No response_

### PR DESCRIPTION
**Pull request by zillani**
add dnsConfig to keycloak statefulset template fixes #16337

### Description of the change

add dnsConfig to keycloak statefulset template fixes #16337

### Benefits

we can add ndots using dnsConfig, this will solve name resolution issues, if ndots >= 5 is used. Example,
keycloak.mynamespace.svc.cluster.local can't be resolved if ndots is >=5. we can update ndots using dnsConfig in statefulset.

### Applicable issues

- fixes #16337

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/keycloak/values.yaml [hunk 1]
```diff
@@ -42,6 +42,20 @@ enableServiceLinks: true
 ## @param commonAnnotations Annotations to add to all deployed objects
 ##
 commonAnnotations: {}
+## @param dnsPolicy DNS Policy for pod
+## ref: https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/
+## E.g.
+## dnsPolicy: ClusterFirst
+dnsPolicy: ""
+## @param dnsConfig DNS Configuration pod
```
Comment (carrodher) [inline bitnami/keycloak/values.yaml:50]:
```suggestion
## @param dnsConfig.ndots DNS Configuration pod
```

**Review by carrodher — COMMENTED** [review]
Thanks for your PR, apart from the inline comments, could you please bump the chart version present in the _Chart.yaml_?

File: bitnami/keycloak/values.yaml (hunk 1 above)
Comment (zillani) [reply bitnami/keycloak/values.yaml:50]: > dnsConfig.ndots

sorry I committed that by mistake while testing, just update like this, dnsConfig: {} & so ndots in the comments are no more required.

**Review by rafariossaa — APPROVED** [review]
LGTM

**Pull request merged**